src/java.base/linux/native/libnio/fs/LinuxNativeDispatcher.c

```diff
@@ -139,6 +140,13 @@ JNIEXPORT void JNICALL
 Java_sun_nio_fs_LinuxNativeDispatcher_endmntent(JNIEnv* env, jclass this, jlong stream)
 {
     FILE* fp = jlong_to_ptr(stream);
-    /* FIXME - man page doesn't explain how errors are returned */
+    // The endmntent() function always returns 1.
     endmntent(fp);
 }
+
+JNIEXPORT jint JNICALL
+Java_sun_nio_fs_LinuxNativeDispatcher_posix_1fadvise(JNIEnv* env, jclass this,
+    jint fd, jlong offset, jlong len, jint advice)
+{
+    return posix_fadvise((int)fd, (off_t)offset, (off_t)len, (int)advice);
```

Comment:
I assume this should posix_fadvise64 as you'll need a off64_t on 32-bit systems.

Reply:
Fixed in b6cb94901d1bbaa34917a5aaf381ecf700282c54.